Makefile

```diff
@@ -14,32 +14,68 @@ E2E_TESTS_NUMBER_PROCESSES_DEFAULT=4  # can be overridden in e2e_tests/.env
+# Command: images
+# Description: Build and push images for API, resource processor and airlock processor.
+# Example: make images
 images: build-and-push-api build-and-push-resource-processor build-and-push-airlock-processor ## 📦 Build and push all images
+
+# Command: build-and-push-api
+# Description: Build and push API image
+# Example: make build-and-push-api
 build-and-push-api: build-api-image push-api-image
+
+# Command: build-and-push-resource-processor
+# Description: Build and push Resource Processor image
+# Example: make build-and-push-resource-processor
 build-and-push-resource-processor: build-resource-processor-vm-porter-image push-resource-processor-vm-porter-image
+
+# Command: build-and-push-airlock-processor
+# Description: Build and push Airlock Processor image
+# Example: make build-and-push-airlock-processor
 build-and-push-airlock-processor: build-airlock-processor push-airlock-processor
 
+# Command: help
+# Description: Display help message on the existing make commands.
+# Example: make help
 help: ## 💬 This help message :)
 	@grep -E '[a-zA-Z_-]+:.*?## .*$$' $(firstword $(MAKEFILE_LIST)) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-25s\033[0m %s\n", $$1, $$2}'
 
-# to move your environment from the single 'core' deployment (which includes the firewall)
-# toward the shared services model, where it is split out - run the following make target before a tre-deploy
-# This will remove + import the resource state into a shared service
+# Command: migrate-firewall-state
+# Description: Migrate the firewall state from the core deployment to the shared services deployment.
+# This is a one-time operation and should only be run when you are moving from the core deployment to the shared services deployment.
+# This command will remove the firewall state from the core deployment and import it into the shared services deployment.
+# Example: make migrate-firewall-state
 migrate-firewall-state: prepare-tf-state
 
+# Command: bootstrap
+# Description: Bootstrap Terraform
+# Example: make bootstrap
 bootstrap:
 	$(call target_title, "Bootstrap Terraform") \
 	&& . ${MAKEFILE_DIR}/devops/scripts/check_dependencies.sh nodocker,env \
 	&& cd ${MAKEFILE_DIR}/devops/terraform && ./bootstrap.sh
 
+# Command: mgmt-deploy
+# Description: Deploy management infrastructure. This will create the management resource group (named <mgmt_resource_group_name> from the config.yaml file) with the necessary resources such as Azure Container Registry, Storage Account for the tfstate and KV for Encryption Keys if enabled.
+# Example: make mgmt-deploy
 mgmt-deploy:
 	$(call target_title, "Deploying management infrastructure") \
 	&& . ${MAKEFILE_DIR}/devops/scripts/check_dependencies.sh nodocker,env \
 	&& cd ${MAKEFILE_DIR}/devops/terraform && ./deploy.sh
 
+# Command: mgmt-destroy
+# Description: Destroy management infrastructure. This will destroy the management resource group with the resources in it.
+# Example: make mgmt-destroy
 mgmt-destroy:
```

Comment:
Do we expect this to be run from the command line directly?
If so, we should add the `##` comment so that it appears when you run `make help`